### PR TITLE
chore(RHTAPWATCH-1137): send messages to sns

### PR DESCRIPTION
**Pull request by avi-biton**
- Add notifier interface
- Add notifier to the reconciler
- Implement SNSNotifier to send messages to Amazon SNS 
- Add mock notifier and use it in integration tests
- Add unit tests

The parameters to communicate with AWS SNS are retrieved from environment variables:

- AWS_ACCESS_KEY_ID
- AWS_SECRET_ACCESS_KEY
- AWS_SESSION_TOKEN (optional)
to create a client
 

In addition, I added:

- NOTIFICATION_REGION for the region
- NOTIFICATION_TOPIC_ARN for the SNS topic ARN

### CODE REVIEW

File: pkg/notifier/notifier.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	snsNotifer := SNSNotifier{
+		SnsClient: sns.NewFromConfig(sdkConfig),
+	}
+
+	return &snsNotifer, nil
+}
+
+func (s *SNSNotifier) Notify(ctx context.Context, message string) error {
```
Comment (yftacherzog) [inline pkg/notifier/notifier.go:50]: We should test this function as well.

Comment (avi-biton) [reply pkg/notifier/notifier.go:50]: We should add unit tests to this file.
For now I only added the Notify test to the integration tests, I will work on the tests on another PR

Comment (yftacherzog) [reply pkg/notifier/notifier.go:50]: I think we're developing bad habits with moving tests to later PRs. 
is there a particular reason not to add the unit tests for this function still with this PR?

File: cmd/main.go [hunk 2]
```diff
@@ -122,9 +123,24 @@ func main() {
...
+		os.Exit(1)
+	}
+
+	ntf := &notifier.SNSNotifier{
+		Pub: snsClient,
+	}
+	if err != nil {
+		setupLog.Error(err, "Error creating notifier")
+		os.Exit(1)
+	}
```
Comment (yftacherzog) [inline cmd/main.go:138]: I think we can delete this.

File: internal/controller/notificationservice_controller.go [hunk 3]
```diff
@@ -76,7 +77,11 @@ func (r *NotificationServiceReconciler) Reconcile(ctx context.Context, req ctrl.
 				logger.Error(err, "Failed to get results", "pipelineRun", pipelineRun.Name)
 				return ctrl.Result{}, err
 			}
-
+			err = r.Notifier.Notify(ctx, string(results))
+			if err != nil {
+				logger.Error(err, "Failed to Notify")
+				return ctrl.Result{}, err
```
Comment (yftacherzog) [inline internal/controller/notificationservice_controller.go:83]: Maybe we can return a re-queue here for now?

File: cmd/main.go (hunk 2 above)
Comment (avi-biton) [reply cmd/main.go]: Right, nice catch

**Review by yftacherzog — APPROVED** [review]
lgtm

File: internal/controller/notificationservice_controller.go (hunk 3 above)
Comment (avi-biton) [reply internal/controller/notificationservice_controller.go]: requeue added

**Review by yftacherzog — APPROVED** [review]

**Pull request merged**